man/J.Rd

```diff
@@ -2,57 +2,58 @@
 \alias{J}
 \alias{CJ}
 \alias{SJ}
-\title{ Creates a Join data table }
+\title{
+  Creates a join \code{data.table}
+}
 \description{
-  Creates a \code{data.table} to be passed in as the \code{i} to a \code{[.data.table} join.
+  Creates a \code{data.table} for use in \code{i} in a \code{[.data.table} join.
 }
 
 \usage{
-# DT[J(\dots)]                           # J() only for use inside DT[\dots].
-SJ(\dots)                                # DT[SJ(\dots)]
-CJ(\dots, sorted = TRUE, unique = FALSE)  # DT[CJ(\dots)]
+DT[J(\dots)]                              # J() only for use inside DT[...].
+SJ(\dots)                                 # DT[SJ(...)]
+CJ(\dots, sorted = TRUE, unique = FALSE)  # DT[CJ(...)]
 }
 
 \arguments{
   \item{\dots}{ Each argument is a vector. Generally each vector is the
-  same length but if they are not then the usual silent repetition is applied. }
-  \item{sorted}{ logical. Should the input be sorted (ascending order)? If \code{FALSE}, the input order is retained. }
+  same length, but if they are not then the usual silent recycling is applied. }
+  \item{sorted}{ logical. Should \code{setkey()} be called on all the columns in the order they were passed to \code{CJ}? }
   \item{unique}{ logical. When \code{TRUE}, only unique values of each vectors are used (automatically). }
 }
 \details{
-    \code{SJ} and \code{CJ} are convenience functions for creating a data.table in the context of a data.table 'query' on \code{x}.
+    \code{SJ} and \code{CJ} are convenience functions to create a \code{data.table} to be used in \code{i} when performing a \code{data.table} 'query' on \code{x}.
 
     \code{x[data.table(id)]} is the same as \code{x[J(id)]} but the latter is more readable. Identical alternatives are \code{x[list(id)]} and \code{x[.(id)]}.
 
-    \code{x} must have a key when passing in a join table as the \code{i}. See \code{\link{[.data.table}}
+    When using a join table in \code{i}, \code{x} must either be keyed or the \code{on} argument be used to indicate the columns in \code{x} and \code{i} which should be joined. See \code{\link{[.data.table}}.
 }
 \value{
   \itemize{
-    \code{J} : the same result as calling list. \code{J} is a direct alias for list but results in clearer more readable code.
+    \code{J} : the same result as calling \code{list}, for which \code{J} is a direct alias.
```

Comment:
I like the note about readability, if not necessarily the exact phrasing. See the recently closed question about why J exists -- here is the place to emphasize why we don't just get rid of in favor of . or list

Reply:
Thanks a lot Jan, Michael and Matt for your additional edits. Obviously I need to learn how to properly make checks also for a seemingly simple edit like this. Sorry for the inconvenience and thanks for your patience and help with my very first PR to data.table.

Reply:
more than happy to help. thank you!

Reply:
no worries @Henrik-P. Most important tricks to learn is `make build && make check` before PR and use of `cc(F)` (it is like build & reload in rstudio). Thanks for investigating your time in data.table!